### PR TITLE
Verificar cabecera de lista de Personal

### PR DESCRIPTION
**Pull request by franzelunico**
image
image

### CODE REVIEW

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java [hunk 1]
```diff
@@ -11,6 +11,9 @@
     @FindBy(xpath = "//button[@id='users']")
     private WebElement usuarioSubMenu;
 
+    @FindBy(xpath = "//button[@id='personal']")
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java]: Chage for use @FindBy(id=...

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java [hunk 2]
```diff
@@ -11,6 +11,9 @@
     @FindBy(xpath = "//button[@id='users']")
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java]: Chage for use @FindBy(id=...

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java [hunk 3]
```diff
@@ -11,6 +11,9 @@
...
     private WebElement usuarioSubMenu;
 
+    @FindBy(xpath = "//button[@id='personal']")
+    private WebElement personalSubMenu;
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java]: Chante. subMenuPersonal

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java [hunk 4]
```diff
@@ -19,4 +22,8 @@ public ListUser selectSubMenuUsuario() {
         CommonEvents.clickButton(usuarioSubMenu);
         return new ListUser();
     }
+
+    public void seleccionarSubMenuPersonal(){
+        CommonEvents.clickButton(personalSubMenu);
+    }
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java]: After click event... return the pageobject that you need for continue your steps

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/ListPersonnel.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    private WebElement tableListPersonnelPhone;
+
+    @FindBy(xpath = "//button[@class='mat-sort-header-button'][contains(text(),'Accion')]")
+    private WebElement tableListPersonnelAction;
+
+    public ListPersonnel() {
+        CommonEvents.isVisible(tableListPersonnel);
+    }
+
+    public boolean validateListPersonnel() {
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/ListPersonnel.java]: Rename validateListPersonnelIsVisible(

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/ListPersonnel.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        return CommonEvents.isVisible(tableListPersonnel);
+    }
+
+    public boolean validateName() {
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/ListPersonnel.java]: Rename validateTableListNameIsVisible

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/ListPersonnel.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        return CommonEvents.isVisible(tableListPersonnelName);
+    }
+
+    public boolean validateLastName() {
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/ListPersonnel.java]: Rename validateTableListLastNameIsVisible

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/ListPersonnel.java [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+        return CommonEvents.isVisible(tableListPersonnelLastName);
+    }
+
+    public boolean validateEmail() {
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/ListPersonnel.java]: Rename validateTableEmailIsVisible

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/ListPersonnel.java [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+        return CommonEvents.isVisible(tableListPersonnelEmail);
+    }
+
+    public boolean validateAddress() {
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/ListPersonnel.java]: Rename validateTableListAdressIsVisible

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/ListPersonnel.java [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+        return CommonEvents.isVisible(tableListPersonnelAddress);
+    }
+
+    public boolean validatePhone() {
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/ListPersonnel.java]: Rename validateTableLisPhoneIsVisible

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/ListPersonnel.java [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+        return CommonEvents.isVisible(tableListPersonnelPhone);
+    }
+
+    public boolean validateAction() {
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/ListPersonnel.java]: Rename validateTableListActionIsVisible

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 12]
```diff
@@ -110,4 +112,46 @@ public void seleccionarSubMenuEquipamiento() throws Throwable {
     public void validarListaDeEquipamientos() throws Throwable {
         Assert.assertTrue(listEquipment.isEquipmentListVisible(), String.format(ErrorMessage.ERROR_MESSAGE_ELEMENT_VISIBLE, "Equipments title"));
     }
-}
\ No newline at end of file
+
+    @And("^click en el submenu 'Personal'$")
+    public void clickEnElSubmenuPersonal() throws Throwable {
+        menuPersonal.seleccionarSubMenuPersonal();
+        Thread.sleep(4000);
```
Comment (jdpapiero) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: Remove this line... is not able use sleep function... add return page object inner method seleccionarSubMenu to solve

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 13]
```diff
@@ -110,4 +112,46 @@ public void seleccionarSubMenuEquipamiento() throws Throwable {
...
+    }
+
+    @Then("^verificar que la tabla lista de 'Personal' este visible$")
+    public void verificarQueLaTablaListaDePersonalEsteVisible() throws Throwable {
+        listPersonnel = new ListPersonnel();
```
Comment (jdpapiero) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: Obtain the page object in previous step in return action

**Review by jdpapiero — CHANGES_REQUESTED** [review]
Review observations

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java [hunk 14]
```diff
@@ -19,4 +22,8 @@ public ListUser selectSubMenuUsuario() {
         CommonEvents.clickButton(usuarioSubMenu);
         return new ListUser();
     }
+
+    public void seleccionarSubMenuPersonal(){
```
Comment (ctola91) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java]: Don't use spanish

**Review by ctola91 — CHANGES_REQUESTED** [review]

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java (hunk 13 above)
Comment (ivanMise) [reply src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: you should get that instance (ListPersonel) 
when doing click in subMenuPersonnel 
maybe you could change  SubMenuPersonnel.java Ej: public ListPerosnel selectSubMenuPersonnel() {
        boolean res = CommonEvents.isVisible(subMenuPersonnel);
        if (res) {
            CommonEvents.clickButton(subMenuPersonnel);
        }
        return new ListPersonnel();
    }

**Review by ctola91 — APPROVED** [review]

**Review by ivanMise — APPROVED** [review]

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 15]
```diff
@@ -743,4 +745,39 @@ public void verificarElTituloDelFormularioDeCreacionNuevoPersonal() throws Throw
         formTitleNewPersonnel = new FormTitleNewPersonnel();
         Assert.assertTrue(formTitleNewPersonnel.validateTitleNewPersonnelIsVisible(), String.format(ErrorMessage.ERROR_MESSAGE_ELEMENT_VISIBLE, "Personnel title"));
     }
-}
+
+    @Then("^verificar que la tabla lista de 'Personal' este visible$")
+    public void verificarQueLaTablaListaDePersonalEsteVisible() throws Throwable {
+        Assert.assertTrue(listPersonnel.validateListPersonnelIsVisible());
```
Comment (marbusjim) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: add error message

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 16]
```diff
@@ -743,4 +745,39 @@ public void verificarElTituloDelFormularioDeCreacionNuevoPersonal() throws Throw
...
+    }
+
+    @And("^verificar que la tabla lista de 'Personal' contenga 'Nombre'$")
+    public void verificarQueLaTablaListaDePersonalContengaNombre() throws Throwable {
+        Assert.assertTrue(listPersonnel.validateNameIsVisible());
```
Comment (marbusjim) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: add error message

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 17]
```diff
@@ -743,4 +745,39 @@ public void verificarElTituloDelFormularioDeCreacionNuevoPersonal() throws Throw
...
+    }
+
+    @And("^verificar que la tabla lista de 'Personal' contenga 'Apellido'$")
+    public void verificarQueLaTablaListaDePersonalContengaApellido() throws Throwable {
+        Assert.assertTrue(listPersonnel.validateLastNameIsVisible());
```
Comment (marbusjim) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: add error message

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 18]
```diff
@@ -743,4 +745,39 @@ public void verificarElTituloDelFormularioDeCreacionNuevoPersonal() throws Throw
...
+    }
+
+    @And("^verificar que la tabla lista de 'Personal' contenga 'Email'$")
+    public void verificarQueLaTablaListaDePersonalContengaEmail() throws Throwable {
+        Assert.assertTrue(listPersonnel.validateEmailIsVisible());
```
Comment (marbusjim) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: add error message

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 19]
```diff
@@ -743,4 +745,39 @@ public void verificarElTituloDelFormularioDeCreacionNuevoPersonal() throws Throw
...
+    }
+
+    @And("^verificar que la tabla lista de 'Personal' contenga 'Direccion'$")
+    public void verificarQueLaTablaListaDePersonalContengaDireccion() throws Throwable {
+        Assert.assertTrue(listPersonnel.validateAddressIsVisible());
```
Comment (marbusjim) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: add error message

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 20]
```diff
@@ -743,4 +745,39 @@ public void verificarElTituloDelFormularioDeCreacionNuevoPersonal() throws Throw
...
+    }
+
+    @And("^verificar que la tabla lista de 'Personal' contenga 'Telefono'$")
+    public void verificarQueLaTablaListaDePersonalContengaTelefono() throws Throwable {
+        Assert.assertTrue(listPersonnel.validatePhoneIsVisible());
```
Comment (marbusjim) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: add error message

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 21]
```diff
@@ -743,4 +745,39 @@ public void verificarElTituloDelFormularioDeCreacionNuevoPersonal() throws Throw
...
+    }
+
+    @And("^verificar que la tabla lista de 'Personal' contenga 'Accion'$")
+    public void verificarQueLaTablaListaDePersonalContengaAccion() throws Throwable {
+        Assert.assertTrue(listPersonnel.validateActionIsVisible());
```
Comment (marbusjim) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: add error message

**Review by marbusjim — COMMENTED** [review]
with comments

**Comment by jdpapiero** [comment]
please review observations, this PR would by closed until fix them

**Pull request closed without merging**